src/xrpld/app/tx/detail/Batch.cpp

```diff
@@ -111,16 +169,20 @@ Batch::preflight(PreflightContext const& ctx)
     // Validation Inner Batch Txns
     std::unordered_set<AccountID> requiredSigners;
     std::unordered_set<uint256> uniqueHashes;
+    std::unordered_map<AccountID, std::unordered_set<std::uint32_t>>
+        accountSequences;
+    std::unordered_map<AccountID, std::unordered_set<std::uint32_t>>
+        accountTickets;
```

Comment:
i don't think we necessarily need two maps. Tickets are really just reserved sequences, and thus Sequence and TicketSequence should never collide for an account. So I think it's actually safer to  compare all the Sequences and TicketSequences using just one map.

perhaps worth adding a unit test where the Sequence collides with TicketSequence

Reply:
The sequence will be 0 when using a ticket which will create duplicates. So I could add some logic to avoid that if we only want one map.

Reply:
yeah, we can use one map if we add only if sequence is zero and ticket is non-zero

Reply:
fixed: https://github.com/XRPLF/rippled/pull/5060/commits/64483176cadaaef4926fb4533e4957f9e7511135